src/pymor/operators/numpy.py

```diff
@@ -396,3 +397,53 @@ def _format_repr(self, max_width, verbosity):
         else:
             matrix_repr = f'<{self.range.dim}x{self.source.dim} dense>'
         return super()._format_repr(max_width, verbosity, override={'matrix': matrix_repr})
+
+
+class NumpyHankelOperator(NumpyGenericOperator):
+    def __init__(self, markov_parameters, source_id=None, range_id=None, solver_options=None, name=None):
+        markov_parameters = np.squeeze(markov_parameters)
+        assert markov_parameters.ndim == 1
+        try:
+            markov_parameters.setflags(write=False)  # make numpy arrays read-only
+        except AttributeError:
+            pass
+
+        self.__auto_init(locals())
+        self.source = NumpyVectorSpace(ceil(len(markov_parameters) / 2), source_id)
+        self.range = NumpyVectorSpace(ceil(len(markov_parameters) / 2), range_id)
```

Comment:
This actually changes the behaviour, because `//` floors the number. I am pretty sure it works correctly at the current stage.

How the inputs are interpreted for the construction of the matrix is a matter of conventions that should be discussed. I appreciate that the implicit construction of the matrix and the conditional zero-padding of the Markov parameters is a bit enigmatic. I will update the PR description with some clarifications.

Reply:
Sorry, I somehow have read 'floor' instead of 'ceil' ..